src/fides/api/main.py

```diff
@@ -92,6 +89,72 @@
 ]
 
 
+def create_fides_app(
+    cors_origins: List[str] = CONFIG.security.cors_origins,
+    routers: List = ROUTERS,
+    app_version: str = VERSION,
+    api_prefix: str = API_PREFIX,
+    request_rate_limit: str = CONFIG.security.request_rate_limit,
+    rate_limit_prefix: str = CONFIG.security.rate_limit_prefix,
+) -> FastAPI:
+    """Return a properly configured application."""
+
+    fastapi_app = FastAPI(title="fides", version=app_version)
+    fastapi_app.state.limiter = Limiter(
+        default_limits=[request_rate_limit],
+        headers_enabled=True,
+        key_prefix=rate_limit_prefix,
+        key_func=get_remote_address,
+        retry_after="http-date",
+    )
+    fastapi_app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
+    for handler in ExceptionHandlers.get_handlers():
+        fastapi_app.add_exception_handler(FunctionalityNotConfigured, handler)
+    fastapi_app.add_middleware(SlowAPIMiddleware)
+
+    if cors_origins:
+        fastapi_app.add_middleware(
+            CORSMiddleware,
+            allow_origins=[str(origin) for origin in cors_origins],
+            allow_credentials=True,
+            allow_methods=["*"],
+            allow_headers=["*"],
+        )
+
+    for router in routers:
+        fastapi_app.include_router(router)
+    fastapi_app.include_router(user_router, tags=["Users"], prefix=f"{api_prefix}")
+    fastapi_app.include_router(api_router)
+
+    return fastapi_app
+
+
+def configure_security_env_overrides(
+    application: FastAPI, security_env: str = CONFIG.security.env.value
+) -> FastAPI:
+    """
+    Configures the FastAPI dependency overrides depending on the
+    value of 'security.env'.
+    """
+
+    if security_env == "dev":
+        # This removes auth requirements for CLI-related endpoints
+        # and is the default
+        application.dependency_overrides[verify_oauth_client_cli] = get_root_client
+    elif security_env == "demo":
+        # This remove all auth requirements
+        application.dependency_overrides[verify_oauth_client] = get_root_client
+        application.dependency_overrides[verify_oauth_client_cli] = get_root_client
+    elif security_env == "prod":
+        # This is the most secure, so all security deps are maintained
+        pass
+    return application
+
+
+app = create_fides_app()
+app = configure_security_env_overrides(application=app)
```

Comment:
should these actually just be a single function since we're doing all other configuration in `create_fides_app`?